clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -4114,27 +4114,28 @@ private Integer nodeFor(ConfigResource resource) {
         }
     }
 
-    private List<MemberIdentity> getMembersFromGroup(String groupId, String reason) {
-        Collection<MemberDescription> members;
-        try {
-            members = describeConsumerGroups(Collections.singleton(groupId)).describedGroups().get(groupId).get().members();
-        } catch (Exception ex) {
-            throw new KafkaException("Encounter exception when trying to get members from group: " + groupId, ex);
-        }
-
-        List<MemberIdentity> membersToRemove = new ArrayList<>();
-        for (final MemberDescription member : members) {
-            MemberIdentity memberIdentity = new MemberIdentity().setReason(reason);
+    private KafkaFutureImpl<List<MemberIdentity>> getMembersFromGroup(String groupId, String reason) {
+        KafkaFutureImpl<List<MemberIdentity>> future = new KafkaFutureImpl<>();
 
-            if (member.groupInstanceId().isPresent()) {
-                memberIdentity.setGroupInstanceId(member.groupInstanceId().get());
+        describeConsumerGroups(Collections.singleton(groupId)).describedGroups().get(groupId).whenComplete((res, ex) -> {
+            if (ex != null) {
+                future.completeExceptionally(new KafkaException("Encounter exception when trying to get members from group: " + groupId, ex));
             } else {
-                memberIdentity.setMemberId(member.consumerId());
+                List<MemberIdentity> membersToRemove = new ArrayList<>();
+                for (final MemberDescription member : res.members()) {
+                    MemberIdentity memberIdentity = new MemberIdentity().setReason(reason);
```

Comment:
If you prefer to use lambda function, please try to eliminate the local variable. for example:`
```java
                    membersToRemove.add(member.groupInstanceId()
                            .map(id -> new MemberIdentity().setGroupInstanceId(id))
                            .orElseGet(() -> new MemberIdentity().setMemberId(member.consumerId()))
                            .setReason(reason));
```

Reply:
Hi @chia7712 , thanks for review.
I rewrite this part, please take a look.